components/Table/Table.js

```diff
@@ -85,7 +96,63 @@ export class Table {
       menuBtn.classList.add('menuBtn');
       menuBtn.textContent = 'more_vert';
 
-      actionsContainer.append(arrowBtn, menuBtn);
+      const dropDownMenu = document.createElement('div');
+      dropDownMenu.classList.add('drop-down__menu');
+      const menuItems = this.menuConfig.map((item) => {
+        const menuItem = document.createElement('a');
+        menuItem.classList.add('menu-item');
+        menuItem.textContent = item.label;
+
+        if (!col.sortable) {
+          if (item.key === 'sortbyasc') {
+            menuItem.style.display = 'none';
+          }
+          if (item.key === 'sortbydesc') {
+            menuItem.style.display = 'none';
+          }
+        }
+
+        menuItem.addEventListener('click', (e) => {
+          const dataKey = cell.getAttribute('data-key');
+
+          switch (item.key) {
+            case 'unsort':
+              this.sortCallback();
+              break;
+            case 'sortbyasc':
+              this.isOrderAsc = true;
+              this.sortCallback(dataKey, this.isOrderAsc);
+              break;
+            case 'sortbydesc':
+              this.isOrderAsc = false;
+              this.sortCallback(dataKey, this.isOrderAsc);
+              break;
+            case 'filter':
+              this.filterMenu();
+              break;
+            case 'hide':
+              cell.style.display = 'none';
+              for (let key in this.tableElementsMap) {
+                if (key === dataKey) {
+                  this.tableElementsMap[key].forEach(
+                    (item) => (item.style.display = 'none')
+                  );
+                }
+              }
```

Comment:
We do not need to have a loop for this map
You have an exact key, values of what you're going to get from `tableElementsMap`
Just get them and hide
`
const dataElements = this.tableElementsMap[dataKey];
if (dataElements) {
  ...your_code_here...
}
`